### PR TITLE
[WIP] Add golang SDK

### PR DESCRIPTION
**Pull request by zhilingc**
Addresses #260 .

Some considerations:
- The protobuf defined request and response payloads are not very user friendly, so we need wrappers around them
- Performance is of great importance for this SDK so we want to avoid reflection as much as possible
- Most feast usecases deal with single-type arrays of features, but there are usecases that involve mixed type outputs, so we have to support both

### CODE REVIEW

**Comment by zhilingc** [comment]
Usage example for the current iteration:
```
client, err := NewGrpcClient("localhost", 6565)
if err != nil {
	panic(err)
}
ctx := context.Background()
response, err := client.GetOnlineFeatures(ctx, &OnlineFeaturesRequest{
	Features:      []string{"fs:1:feature1", "fs:1:feature2"},
	MaxAgeSeconds: 100,
	Entities: []Row{
		{"s2id": Int64Val(1)},
		{"s2id": Int64Val(2)},
	},
})
if err != nil {
	panic(err)
}

// just retrieve rows
// returns map[string]*types.Value
_ = response.Rows()

// use convenience functions to get arrays
// returns [][]int64
_, err = response.Int64Arrays([]string{"fs:1:feature1", "fs:1:feature2"}, []int64{-1, -1})
```

File: sdk/go/request.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+type OnlineFeaturesRequest struct {
+
+	// Features is the list of features to obtain from Feast. Each feature must be given by its fully qualified ID,
+	// in the format featureSet:version:featureName.
+	Features      []string
+
+	// MaxAgeSeconds is the maximum allowed staleness of the features in seconds. This max age will be applied to all
+	// featureSets in this request.
+	// Setting this value to 0 will cause feast to default to the max age specified on the feature set spec, if any.
+	MaxAgeSeconds int
```
Comment (woop) [inline sdk/go/request.go]: Should we leave this out for now? Users can request features from multiple feature sets, and they will definitely have different maximum ages. So IMHO either we need a way to set this at the feature set level, or we should just leave it to default (registered max age).

File: sdk/go/request.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+func (r OnlineFeaturesRequest) buildRequest() (*serving.GetOnlineFeaturesRequest, error) {
+	featureSets, err := buildFeatureSets(r.Features, r.MaxAgeSeconds)
+	if err != nil {
+		return nil, err
+	}
+	entityRows := make([]*serving.GetOnlineFeaturesRequest_EntityRow, len(r.Entities))
+
+	for i := range r.Entities {
+		entityRows[i] = &serving.GetOnlineFeaturesRequest_EntityRow{
+			EntityTimestamp: &timestamp.Timestamp{Seconds: time.Now().Unix()},
```
Comment (woop) [inline sdk/go/request.go]: We should keep these timestamps consistent with other online clients. In the Python SDK I am not setting this. What is the use case for this specific timestamp? If it is for tracking round trip time then I think that should be external to the entity row.

File: sdk/go/request.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+}
+
+func buildFeatureSets(features []string, maxAgeSeconds int) ([]*serving.GetOnlineFeaturesRequest_FeatureSet, error) {
+	featureSetMap := map[string]*serving.GetOnlineFeaturesRequest_FeatureSet{}
+	for _, feature := range features {
+		split := strings.Split(feature, ":")
+		if len(split) != 3 {
+			return nil, fmt.Errorf(ErrInvalidFeatureName, feature)
+		}
+		key := split[0] + ":" + split[1]
```
Comment (kaihendry) [inline sdk/go/request.go]: Could do a `featureSet, version, featureName := split[0], split[1], split[2]` like https://stackoverflow.com/a/16551613/4534 to be more explicit

Comment (zhilingc) [reply sdk/go/request.go]: oh, nice, thanks!

File: sdk/go/request.go (hunk 2 above)
Comment (zhilingc) [reply sdk/go/request.go]: Right. I forgot that feast itself does set the timestamp to the current time if not provided by the client.

File: sdk/go/request.go (hunk 1 above)
Comment (zhilingc) [reply sdk/go/request.go]: Ok. We need to KIV though

**Pull request merged**